Assets/Scripts/Entity/Enemy/MovementBehaviors/EnemyMovement.cs

```diff
@@ -32,7 +32,11 @@ public void MoveTo(Vector3 destination)
 
     public void Stop()
     {
-        agent.isStopped = true;
+        if(agent.enabled)
+        {
+            agent.isStopped = true;
+            agent.enabled = false;
```

Comment:
If we are disabling the agent here then there is no point in stopping it (`isStopped = true`). Though, this function was specifically made so that it would only stop and specifically NOT disable the NavMeshAgent. Reason for this was that if a SetDestination is called on a disabled NavMeshAgent, then it throws an exception. So this may cause errors/bugs to appear elsewhere. 

There is also a warning that occurs when the navmeshobstacle is enabled, stating that both the navmeshagent and navmeshobstacle are enabled. I think this is only happening on the same frame, so it's probably safe to ignore the warning. Unless you can come up with a better placement for the NavMeshObstacle code (As it is now, the Enemy.cs has become dependent on the enemy having a NavMeshObstacle).

Reply:
I don't see why disabling the `NavMeshAgent` would cause any problems if we're not calling `SetDestination` on dead enemies. I've tested this and haven't run into any exceptions.

Yeah, you can ignore that warning; the NavMeshAgent is disabled before the obstacle is enabled, so they won't ever be active at the same time (both occur within the same frame, but one precedes the other, so Unity probably still wanted to warn us).